packages/fetch-engine/src/download.ts

```diff
@@ -291,19 +291,8 @@ async function binaryNeedsToBeDownloaded(
     }
   }
 
-  // If there is no cache and the file doesn't exist, we for sure need to download it
-  if (!targetExists) {
```

Comment:
This would cause the following issue:
- If you have an engine file already downloaded
- This check will be skipped and we continue below